docs/developer-guide/contributing.md

```diff
@@ -291,3 +291,9 @@ The final step is running the End-to-End testsuite, which makes sure that your K
 * When all components have started, run `make test-e2e-local` to run the end-to-end tests against your local services.
 
 For more information about End-to-End tests, refer to the [End-to-End test documentation](test-e2e.md).
+
+
+## Enhancement proposals
+
+If you are proposing a major feature, change in design or process refactor, please help define how it would look like with a new enhancement proposal as described in the enhancement proposal [template](/docs/proposals/001-proposal-template.md).
```

Comment:
Can you help me understand which feature should be considered major? 
I think [Plugins Refactoring](https://docs.google.com/document/d/1xlC9URq8H8ABcvDOjCD8F0HYrmKRsGrCrOmnxRNX9Go/edit) is a major feature. The [Pods View](https://github.com/argoproj/argo-cd/pull/5091) is another example.

Maybe we should add these examples in the doc for future reference?

Reply:
> If a change:
Involves support for a new toolset.
Involves changes in code/project/library dependencies which impact long term maintenance.
Impacts how ArgoCD is operated including changes in capabilities.
Requires consensus/code across the user and contributor community.

The description in the PR addresses that, I should probably add that to the doc?

> I think Plugins Refactoring is a major feature. The Pods View is another example.

Correct!
In general, one or more maintainers / project leads should be making that decision on what needs to be an enhancement proposal and what needn't be.